src/main/kotlin/org/rust/ide/utils/StructFieldsExpander.kt

```diff
@@ -0,0 +1,84 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.utils
+
+import com.intellij.psi.PsiElement
+import com.intellij.psi.impl.source.tree.LeafPsiElement
+import org.rust.lang.core.psi.*
+import org.rust.lang.core.psi.ext.*
+import org.rust.stdext.nextOrNull
+
+fun expandStructFields(
+    factory: RsPsiFactory,
+    rsPat: RsPat,
+    declarationFieldPositions: HashMap<String, Int>,
+    missingFields: List<PsiElement>,
+    hasTrailingComma: Boolean = false
+) {
+    if (rsPat is RsPatStruct) {
+        val patFieldListIterator = rsPat.patFieldList.iterator()
+        if (!patFieldListIterator.hasNext()) {
+            fillRsPatWithFields(factory, rsPat, missingFields, hasTrailingComma)
+            return
+        }
+        var anchor = patFieldListIterator.next()
+        val missingFieldsIterator = missingFields.map { it as RsPatField }.iterator()
+        var insertedFieldsAmount = 0
+        while (missingFieldsIterator.hasNext()) {
+            val missingField = missingFieldsIterator.next()
+            val missingFieldPosition = declarationFieldPositions[missingField.kind.fieldName]!!
+            while (anchor != null) {
+                val anchorPosition = declarationFieldPositions[anchor.kind.fieldName]
+                if (anchorPosition != null && missingFieldPosition < anchorPosition) {
+                    rsPat.addBefore(missingField, anchor)
+                    rsPat.addAfter(factory.createComma(), anchor.getPrevNonCommentSibling())
+                    insertedFieldsAmount++
+                    break
+                } else {
+                    anchor = patFieldListIterator.nextOrNull()
+                }
+            }
+        }
+        fillRsPatWithFields(factory, rsPat, missingFields.drop(insertedFieldsAmount), hasTrailingComma)
+    } else if (rsPat is RsPatTupleStruct) {
+        fillRsPatWithFields(factory, rsPat, missingFields, hasTrailingComma)
+    }
+}
+
+fun getTupleStructMissingFields(factory: RsPsiFactory, amount: Int): List<RsPatBinding> {
```

Comment:
Let's make it private somehow (see subsequent comments) or move it to `RsPsiFactory` and rename with `create` prefix.